### PR TITLE
refactor(connector): [CYBERSOURCE] Remove default case for Cybersource

### PR DESCRIPTION
**Pull request by TejasMate**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

https://github.com/juspay/hyperswitch/issues/2268

## How did you test it?

Make a mandate payment for cybersource, for the payment method  which is not implemented like bank_debit or bank_transfer and check for the error message - it should be PM not implemented

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by SanchithHegde** [comment]
@TejasMate Could you address the failing CI checks?

**Review by swangi-kumari — DISMISSED** [review]

**Review by SamraatBansal — DISMISSED** [review]

**Comment by swangi-kumari** [comment]
Hey @TejasMate ,
Can you please address the default case for Refund Status (line 316) and make the required changes there also.

**Comment by TejasMate** [comment]
> Hey @TejasMate , Can you please address the default case for Refund Status (line 316) and make the required changes there also.

sure

File: crates/router/src/connector/cybersource/transformers.rs [hunk 1]
```diff
@@ -313,7 +326,7 @@ impl From<CybersourcePaymentStatus> for enums::RefundStatus {
                 Self::Success
             }
             CybersourcePaymentStatus::Failed => Self::Failure,
-            _ => Self::Pending,
+            CybersourcePaymentStatus::Pending => Self::Pending,
```
Comment (swangi-kumari) [inline crates/router/src/connector/cybersource/transformers.rs:329]: Hey @TejasMate , you have not fill all the matching arms of `CybersourcePaymentStatus` . Please check it once again and do the required changes.

**Review by swangi-kumari — CHANGES_REQUESTED** [review]

**Comment by swangi-kumari** [comment]
Hey @TejasMate ,
Before pushing your commit please address the failing CI checks
Run `cargo clippy` and addressed lints thrown by it.

**Comment by TejasMate** [comment]
hey @SanchithHegde @swangi-kumari @SamraatBansal ,

I'm not sure where to put these values **Voided, Reversed, Authorized** into the which case: **Success, Pending or Failure** for RefundStatus?

**Comment by swangi-kumari** [comment]
Hey @TejasMate ,
You have to map all the default cases to `Pending` RefundStatus.

**Comment by swangi-kumari** [comment]
Hey @TejasMate ,
Thanks for your interest in contributing to hyperswitch.
Let us know if you need any assistance from our end.
Also, even if hacktoberfest is over, we should celebrate open source everyday and we are open for more contributions from you.
We would still be rewarding folks with goodies even if the PR gets merged post hacktoberfest.
May the Source be with you!

**Review by ArjunKarthik — APPROVED** [review]

**Review by swangi-kumari — APPROVED** [review]

**Pull request merged**